### DISCUSSION TITLE
how we can clean bucket using lifecycle configuration

### DISCUSSION THREAD

**Discussion by ibrahemw** [noobaa/noobaa-core#6771] [Q&A]
i am trying to expire object by using lifecycle configuration, is it possible? is there a CLI command exists in noobaa or we need to use S3 APIs to do it ?

**Comment by guymguym** [comment]
Hi @ibrahemw 
Use S3 API to set up lifecycle - for example using aws cli - see example below.

!!! ----- !!! HOWEVER please notice we just found an issue that uses the expiration days as minutes - https://github.com/noobaa/noobaa-core/issues/6818 which means you're going to have to multiply the desired value by24*60 to compensate for that...

https://docs.aws.amazon.com/cli/latest/reference/s3api/put-bucket-lifecycle-configuration.html

```
aws --endpoint $ENDPOINT \
  s3api put-bucket-lifecycle-configuration \
  --bucket my-bucket \
  --lifecycle-configuration '{
    "Rules": [{
      "ID": "expire-objects",
      "Status": "Enabled",
      "Prefix": "tmp/autoexpire/",
      "Expiration": { "Days": '$((30*24*60))' }
    }]
  }'
```
